### PR TITLE
No unused proptypes

### PR DESCRIPTION
**Pull request by oponder**
Inspired by: https://github.com/giantswarm/happa/pull/1668

I thought maybe it's possible to autofix unused props. But that doesn't seem to happen.

### CODE REVIEW

**Comment by axbarsan** [comment]
image

**Comment by ubergesundheit** [comment]
What @axbarsan said ;)

File: src/components/Cluster/NewCluster/NewClusterWrapper.tsx [hunk 1]
```diff
@@ -78,7 +78,7 @@ const NewClusterWrapper: FC<INewClusterWrapperProps> = ({
 
   const creationCapabilities = useMemo(
     () => makeCapabilities(selectedRelease, provider),
-    [selectedRelease, provider]
+    [selectedRelease, provider, makeCapabilities]
```
Comment (oponder) [inline src/components/Cluster/NewCluster/NewClusterWrapper.tsx:81]: Fixes an unrelated warning.

File: src/components/shared/VirtualizedScrollableGrid.js [hunk 2]
```diff
@@ -76,19 +76,20 @@ class VirtualizedScrollableGrid extends React.PureComponent {
   }
 
   updateResponsiveFeatures = debounce(() => {
+    const { theme, columnCount, adaptWidthToElement, width } = this.props;
```
Comment (oponder) [inline src/components/shared/VirtualizedScrollableGrid.js:79]: destructured it here so the linter can see it being used, also kinda nice for the humans, you can see what `setColumnCount` and `setWidth` actually use from the props directly.

Possibly a `fmt` thing: try to limit (or be more explicit) about the amount of things being passed around, when a function only needs a subset of the object.

File: src/components/Home/UpgradeNotice.tsx [hunk 3]
```diff
@@ -38,11 +38,13 @@ function UpgradeNotice({
   canClusterUpgrade,
   onClick,
   className,
+  clusterId,
 }: IUpgradeNoticeProps) {
   if (!canClusterUpgrade) return null;
 
   return (
     <UpgradeWrapperDiv
+      id={`upgrade-notice-${clusterId}`}
```
Comment (oponder) [inline src/components/Home/UpgradeNotice.tsx:47]: it warned about clusterId not being used, but it most certainly is being used in `mapStateToProps`

So I added it here as an id to kind of trick the linter...  not too happy about that. Any thoughts?

Comment (ubergesundheit) [reply src/components/Home/UpgradeNotice.tsx:47]: Thats a perfectly valid fix :+1:

**Pull request merged**